proposal.md

```diff
@@ -1,20 +1,23 @@
-Name: ____________           ID:   ____________
+Name: Pritheesh Panchmahalkar
+ID:   47524741
 
 ## Proposed Project
 
-> Replace this paragraph with the description of what your project
-> will do. Tell me what kind of interface it will have, and roughly
-> how it will work. I don't need more than three or four sentences,
-> unless you feel compelled to write more.
+Digital Signature
+
+I would like to develop a console based application which implements a digital signature. 
+
+Sender uses the services of the server to generate a digital signature and sends it to a receiver.
+
+The receiver then uses the services of the server again and checks for the integrity and authentication of the message received.
 
-## Outline Structure
 
-> Describe how you'll organize your code. What is the process and
-> supervision structure? If it uses a framework, how does it fit in. I
-> just need to understand the direction you plan to take. Again, three
-> or four sentences are probably enough.
 
 
+## Outline Structure
+
+An application that stores/generates the keys for encryption and decryption. (acts like a CA)
+
+Another application which acts as a server that uses the above application for keys and does the encryption, decryption and hashing
 
-> replace all the ">" lines with your content, then push this to
-> github and issue a merge request.
+A sender and receiver use the server to generate digital signatures and check for integrity and authentication of message sent.
```

Comment:
Could you talk more about servers and supervisors? I need to know what the application will look like, where you store state, and so on.

Reply:
The two applications are similar to hangman and dictionary. One application has an agent which stores the state which is supervised. Another application uses the state and provides API for the clients. This application is similar to hangman.